src/app/common/components/MustGatherWatcher.tsx

```diff
@@ -44,10 +46,26 @@ export const MustGatherWatcher: React.FunctionComponent<IMustGatherWatcherProps>
           <div>
             You can download the migration logs for the {type}&nbsp;
             <Button
-              component="a"
-              href={getMustGatherApiUrl(`must-gather/${data?.['id']}/data`)}
-              download
-              target="_self"
+              onClick={() => {
+                authorizedFetch<Blob>(
```

Comment:
We could probably abstract these two fetch-then-save code snippets into a helper function, but it's not too much code so I'm fine either way.

Reply:
Not a problem, I went ahead and created a few helper functions for fetching, downloading, and complaining about the must gather output file. I've realized an issue with the notifications for when there are many at a single time, but we can follow up on that outside of this PR.